plugins/mediatek-scaler/fu-mediatek-scaler-device.c

```diff
@@ -0,0 +1,849 @@
+			    "invalid read buffer, checksum expected 0x%02x, got 0x%02x.",
+			    checksum,
+			    buf[report_data_sz + 2]);
+		return FALSE;
+	}
+
+	/* checksum is needn't */
+	g_byte_array_append(readbuf, buf, report_data_sz + 2);
+
+	/* print the raw data */
+	rawbytes = g_bytes_new(readbuf->data, (gsize)readbuf->len);
+	fu_dump_bytes(G_LOG_DOMAIN, "DDC/CI read report", rawbytes);
+	return TRUE;
+}
+
+static gboolean
+fu_mediatek_scaler_device_set_ddc_priority(FuDevice *device, guint8 flag, GError **error)
+{
+	FuMediatekScalerDevice *self = FU_MEDIATEK_SCALER_DEVICE(device);
+	g_autoptr(GByteArray) cod = g_byte_array_new();
+	g_autoptr(GError) error_local = NULL;
+	const guint8 buf[2] = {DDC_OPCODE_MTK_REQ, DDC_VCP_CODE_PRIORITY};
+
+	g_byte_array_append(cod, buf, sizeof(buf));
+	g_byte_array_append(cod, &flag, sizeof(guint8));
+	if (!(mediatek_scaler_device_ddc_write(self, cod, &error_local))) {
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_NOT_SUPPORTED,
+			    "failed to set priority '0x%u', unsupported display? %s",
+			    flag,
+			    (error_local != NULL ? error_local->message : "unknown"));
+		fu_device_sleep(device, 10);
+		return FALSE;
+	}
+	fu_device_sleep(device, 10);
+	return TRUE;
+}
+
+static gboolean
+fu_mediatek_scaler_display_is_connected(FuDevice *device, GError **error)
+{
+	FuMediatekScalerDevice *self = FU_MEDIATEK_SCALER_DEVICE(device);
+	g_autoptr(GByteArray) cmdbuf = g_byte_array_new();
+	g_autoptr(GByteArray) readbuf = g_byte_array_new();
+	g_autoptr(GError) error_local = NULL;
+	guint8 randval1 = g_random_int_range(1, 255);
+	guint8 randval2 = g_random_int_range(1, 255);
+	const guint8 cmd[] = {DDC_OPCODE_MTK_REQ, DDC_VCP_CODE_SUM, randval1, randval2};
+
+	g_byte_array_append(cmdbuf, cmd, sizeof(cmd));
+	if (mediatek_scaler_device_ddc_read(self, cmdbuf, readbuf, &error_local)) {
+		/* device unique feature */
+		if (readbuf->data[3] == (randval1 + randval2)) {
+			g_info("found mediatek display controller: %s, i2c-dev: %s",
+			       self->dp_aux_dev_name,
+			       self->i2c_dev_name);
+			return TRUE;
+		}
+	}
+	g_set_error(error,
+		    FWUPD_ERROR,
+		    FWUPD_ERROR_NOT_SUPPORTED,
+		    "unsuccessful feature test, could be unsupported display: %s",
+		    (error_local != NULL ? error_local->message : "unknown"));
+	return FALSE;
+}
+
+static gboolean
+fu_mediatek_scaler_display_is_connected_cb(FuDevice *device, gpointer user_data, GError **error)
+{
+	return fu_mediatek_scaler_display_is_connected(device, error);
+}
+
+static gchar *
+mediatek_scaler_device_probe_hardware_version(FuDevice *device, GError **error)
+{
+	FuMediatekScalerDevice *self = FU_MEDIATEK_SCALER_DEVICE(device);
+	g_autoptr(GByteArray) cmdbuf = g_byte_array_new();
+	g_autoptr(GByteArray) readbuf = g_byte_array_new();
+	const guint8 cmd[] = {DDC_OPCODE_MTK_REQ, DDC_VCP_CODE_VERSION, 0x00};
+	g_autoptr(FuDeviceLocker) locker = NULL;
+
+	/* open device */
+	if ((locker = fu_device_locker_new(self->i2c_dev, error)) == NULL)
```

Comment:
nit; I don't really like combining these two lines.  Can you please split them?
```
locker = fu_device_locker_new();
if (locker == NULL)
```

Reply:
Fixed.